core/src/main/kotlin/com/justai/jaicf/BotEngine.kt

```diff
@@ -63,12 +65,19 @@ class BotEngine(
     private val conversationLoggers: Array<ConversationLogger> = arrayOf(Slf4jConversationLogger())
 ) : BotApi, WithLogger {
 
-    private val activators = activators.map { it.create(model) }.let {
-        val catchAllActivator = CatchAllActivator.create(model)
-        if (it.none { a -> a.name ==  catchAllActivator.name }) {
-            it + catchAllActivator
-        } else {
-            it
+    private val activators = activators.map { it.create(model) }.addBuiltinActivators()
+
+    private fun List<Activator>.addBuiltinActivators(): List<Activator> {
+        fun MutableList<Activator>.removeIfPresent(a: Activator) = removeIf { it.name == a.name }
+        fun MutableList<Activator>.addIfAbsent(a: Activator) = find { it.name == a.name } ?: add(a)
+
+        val catchAll = CatchAllActivator.create(model)
+        val builtinActivators = listOf(BaseEventActivator, BaseIntentActivator).map { it.create(model) }
```

Comment:
Why not to add `CatchAllActivator` here and try to remove if present all of these activators?

Reply:
Basically, I don't want to force our order if the user provides activators by himself in a certain order, and I'd better do `addIfAbsent` for all of these activators without rearranging them.
But what if the user puts `CatchAllActivator` in the middle of the list? Where then `BaseEvent` and `BaseIntent` should go, before or after `CatchAll`?
So, I decided to consider `CatchAllActivator` as a special one that must always be present and must always be at the end of the list. At the same time, other activators will go to the end (before `CatchAll`) only if the user hasn't specified the order directly.